### PR TITLE
test: mailbox

### PR DESCRIPTION
**Pull request by steve806**
* closes MITO-15

### CODE REVIEW

**Review by hashableric — CHANGES_REQUESTED** [review]
please remove unused warning

File: contracts/mailbox/src/state.rs [hunk 1]
```diff
@@ -25,3 +27,77 @@ pub const MESSAGE_TREE: Item<MerkleTree> = Item::new(MESSAGE_TREE_KEY);
 
 pub const MESSAGE_PROCESSED_PREFIX: &str = "message_processed";
 pub const MESSAGE_PROCESSED: Map<Vec<u8>, bool> = Map::new(MESSAGE_PROCESSED_PREFIX);
+
+pub fn assert_owner(owner: Addr, sender: Addr) -> Result<(), ContractError> {
+    if owner != sender {
```
Comment (byeongsu-hong) [inline contracts/mailbox/src/state.rs:32]: i suggest to use pointer instead of moving values

File: contracts/mailbox/src/state.rs [hunk 2]
```diff
@@ -25,3 +27,77 @@ pub const MESSAGE_TREE: Item<MerkleTree> = Item::new(MESSAGE_TREE_KEY);
...
+        return Err(ContractError::InvalidDestinationDomain {
+            domain: message_dest_domain,
+        });
+    }
+
+    Ok(())
+}
+
+pub fn assert_already_delivered(storage: &dyn Storage, id: Binary) -> Result<(), ContractError> {
+    if !MESSAGE_PROCESSED.may_load(storage, id.0.clone())?.is_none() {
```
Comment (byeongsu-hong) [inline contracts/mailbox/src/state.rs]: ask; it seems working ok with `&id.0` can you check it?

Comment (steve806) [reply contracts/mailbox/src/state.rs]: it says when change `id.0.clone()` to `&id.0`

```
mismatched types
expected struct `std::vec::Vec<u8>`
found reference `&std::vec::Vec<u8>`
```

File: contracts/mailbox/src/state.rs (hunk 1 above)
Comment (steve806) [reply contracts/mailbox/src/state.rs:32]:
```
pub fn assert_owner(owner: &Addr, sender: &Addr) -> Result<(), ContractError> {
    if owner != sender {
```
like this?

Comment (byeongsu-hong) [reply contracts/mailbox/src/state.rs:32]: Yes!

Comment (byeongsu-hong) [reply contracts/mailbox/src/state.rs:32]: Yes!

File: contracts/mailbox/src/state.rs (hunk 2 above)
Comment (byeongsu-hong) [reply contracts/mailbox/src/state.rs]: oh i see

File: contracts/mailbox/src/core.rs [hunk 3]
```diff
@@ -169,10 +169,10 @@ mod tests {
         )
         .unwrap_err();
 
-        let len = long_recipient_address.clone().len();
+        let _len = long_recipient_address.clone().len();
         assert!(matches!(
             invalid_address_length_assert,
-            ContractError::InvalidAddressLength { len }
+            ContractError::InvalidAddressLength { len: _len }
```
Comment (byeongsu-hong) [inline contracts/mailbox/src/core.rs]: How about use long_recipient_address.len() directly?

File: contracts/mailbox/src/core.rs [hunk 4]
```diff
@@ -201,11 +201,11 @@ mod tests {
         )
         .unwrap_err();
 
-        let version = wrong_decoded_message.version;
+        let _version = wrong_decoded_message.version;
 
         assert!(matches!(
             invalid_message_version_assert,
-            ContractError::InvalidMessageVersion { version }
+            ContractError::InvalidMessageVersion { version: _version }
```
Comment (byeongsu-hong) [inline contracts/mailbox/src/core.rs]: Here too

**Review by byeongsu-hong — CHANGES_REQUESTED** [review]

**Review by byeongsu-hong — APPROVED** [review]

**Review by hashableric — APPROVED** [review]

**Pull request merged**